src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts

```diff
@@ -8,101 +8,385 @@ import {
+import "../../../../../components/ha-svg-icon";
+import {
+  AttributeConfigurationStatus,
+  Cluster,
+  ClusterConfigurationEvent,
+  ClusterConfigurationStatus,
+  fetchClustersForZhaNode,
+  reconfigureNode,
+  ZHA_CHANNEL_CFG_DONE,
+  ZHA_CHANNEL_MSG_BIND,
+  ZHA_CHANNEL_MSG_CFG_RPT,
+} from "../../../../../data/zha";
 import { fireEvent } from "../../../../../common/dom/fire_event";
+import { computeRTLDirection } from "../../../../../common/util/compute_rtl";
 
 @customElement("dialog-zha-reconfigure-device")
 class DialogZHAReconfigureDevice extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @internalProperty() private _active = false;
+  @internalProperty() private _status?: string;
 
-  @internalProperty() private _formattedEvents = "";
+  @internalProperty() private _stages?: string[];
 
-  @internalProperty()
-  private _params: ZHAReconfigureDeviceDialogParams | undefined = undefined;
+  @internalProperty() private _clusterConfigurationStatuses?: Map<
+    number,
+    ClusterConfigurationStatus
+  > = new Map();
 
-  private _subscribed?: Promise<() => Promise<void>>;
+  @internalProperty() private _params:
+    | ZHAReconfigureDeviceDialogParams
+    | undefined = undefined;
+
+  @internalProperty() private _allSuccessful = true;
 
-  private _reconfigureDeviceTimeoutHandle: any = undefined;
+  @internalProperty() private _showDetails = false;
+
+  private _subscribed?: Promise<() => Promise<void>>;
 
   public async showDialog(
     params: ZHAReconfigureDeviceDialogParams
   ): Promise<void> {
     this._params = params;
-    this._subscribe(params);
+    this._stages = undefined;
   }
 
   public closeDialog(): void {
     this._unsubscribe();
-    this._formattedEvents = "";
     this._params = undefined;
+    this._status = undefined;
+    this._stages = undefined;
+    this._clusterConfigurationStatuses = undefined;
     fireEvent(this, "dialog-closed", { dialog: this.localName });
   }
 
   protected render(): TemplateResult {
     if (!this._params) {
       return html``;
     }
+
     return html`
       <ha-dialog
         open
-        hideActions
-        @closing="${this.closeDialog}"
-        .heading=${createCloseHeading(
-          this.hass,
-          this.hass.localize(`ui.dialogs.zha_reconfigure_device.heading`)
-        )}
+        @closed="${this.closeDialog}"
+        .heading=${html`
+          <span class="header_title">
+            ${this.hass.localize(`ui.dialogs.zha_reconfigure_device.heading`)}:
+            ${this._params?.device.user_given_name || this._params?.device.name}
+          </span>
+          <mwc-icon-button
+            aria-label=${this.hass.localize("ui.dialogs.generic.close")}
+            dialogAction="close"
+            class="header_button"
+            dir=${computeRTLDirection(this.hass)}
+          >
+            <ha-svg-icon .path=${mdiClose}></ha-svg-icon>
+          </mwc-icon-button>
+        `}
```

Comment:
Why not use `createCloseHeading`?

Reply:
I think I had something else in there at some point... I had the spinner in it that's what it was. I can fix this

Reply:
done